examples/example_matching.py

```diff
@@ -50,10 +62,27 @@
 # -----------------------------------------------------------------------------
 # Step 2: Clean the data/feature engineering to prepare for matching model
 # -----------------------------------------------------------------------------
+run_options = RunOptions(
+    pretty_print_sql=True, debug_mode=True, debug_show_sql=True, debug_incremental=True
+)
 
 df_ch_clean = clean_data_using_precomputed_rel_tok_freq(df_ch, con=con)
 df_fhrs_clean = clean_data_using_precomputed_rel_tok_freq(df_fhrs, con=con)
 
+# -----------------------------------------------------------------------------
+# Step 2.5: Run exact matching to reduce the number of records to consider
+# -----------------------------------------------------------------------------
+
+exact_match_results = run_deterministic_match_pass(
+    con=con,
+    df_addresses_to_match=df_fhrs_clean,
+    df_addresses_to_search_within=df_ch_clean,
+)
+
+exact_match_summary = calculate_exact_match_metrics(exact_match_results)
+print("\nExact match results summary:")
+print(exact_match_summary)
```

Comment:
Do we then want to remove these exact matches from `df_fhrs_clean`?

Reply:
one option is that the next step optionally accepts a dataframe containing the answer from the first step.  If provided, it combines results.